### ISSUE TITLE
Bug Report: Enabling replication heartbeat on existing cluster makes replica go unhealthy (not serving)

### ISSUE THREAD

**Issue by vkovacik** [vitessio/vitess#12065]
### Overview of the Issue

**Use Case:**
Enable replication tracking using heatbeat (heartbeat_enable=True) on an existing cluster.

**Background:**
Replication heartbeat is tracked via table `_vt.heartbeat` which is created by the primary and replicated to the replicas. Replica reads heartbeat (timestamp) from the table and reports replication lag.

**Problem:**
In Kubernetes + operator + vtorc setup, operator can apply the change (heartbeat_enable=True) on replica as the first tablet in the shard. This makes replica to fail and go unhealthy (not serving) because replication HeartbeatReader will fail reading from missing heartbeat table.

vitess-operator applies shard (vttablet) config change in non-deterministic order so we must assume worst-case that it starts with a replica. When a replica goes unhealthy, the operator will block updates to the remaining tablet in shard so and it will never get to updating the primary.

Even in a lucky case, when the primary is updated as the first tablet in the shard, vtorc will detect downed primary and promote a replica that has not yet been updated and is running with heartbeat mode still disabled. This will again break the remaining replicas.

**Suggestion:**
Make replication HeartbeatReader ignore missing table and return faked "healthy" status. This will make replicas to continue being healthy during update process. vitess-operator will be able to finish shard config update process, eventually getting to the primary which will create the heartbeat table.

### Reproduction Steps

Deployment method: Kubernetes + vitess-operator + vtorc

1. Build cluster with vttablet `heartbeat_enable=False` (default)
2. Update vttablet config with `heartbeat_enable=True`
3. vitess-operator applies config to tablets in the shard in non-deterministic order
4. Let's assume that replica is updated first
5. Replica will go unhealthy (not serving)
6. vitess-operator will block further shard tablet updates so it will never get to the primary

### Binary Version

```sh
Vitess version 14.0.3
```

### Operating System and Environment details

```sh
N/A
```

**Comment by vkovacik** [comment]
This has been discussed with @mattlord in https://vitess.slack.com/archives/C0PQY0PTK/p1673377463143619

**Comment by deepthi** [comment]
The last time we visited this issue, here's what was proposed, but never implemented.
https://github.com/vitessio/vitess/pull/6665#issuecomment-701634139

This will be a better solution than faking healthy (which can cause other problems).

**Comment by deepthi** [comment]
@vkovacik did you ever manage to work around this?

**Comment by bluecrabs007** [comment]
> [@vkovacik](https://github.com/vkovacik) did you ever manage to work around this?

No, we still have this issue and once a cluster is created without heartbeat enabled, it is not possible to enable it.

**Comment by deepthi** [comment]
I can tell you how we worked around it.
- manually insert/update the single row on _vt.heartbeat table on the primary with current timestamp. make sure it has replicated to all replicas
- this takes care of the replica lag problem
- add flags to vttablet extraFlags in your yaml and roll it out immediately

This works on v16+ because we implemented schema initialization such that _vt internal tables are always created even if the functionality is disabled.
